pimd: Querier to non-querier transition to be ignored (ANVL Conformance fix)

mobash-rasool opened the pull request:
Fixing IGMPv2 ANVL Conformance 3.10

As per RFC 2236 section 3, when the leave message is received at a querier,
it starts sending Query messages for "last Member Query Interval*query count"
During this time there should not be any querier to non-querier
transition and the same router needs to send the remaning queries.

Currently the code is handling this scenario only when leave is receive
for a group and the query is received for the same group.
But we need to handle it irrespective of group since the querier
election is based on interface and not group.

Signed-off-by: Mobashshera Rasool <mrasool@vmware.com>

donaldsharp commented:
I hate the wording in the rfc.  I could interpret it either way.  Asking around for opinions here.

donaldsharp commented:
I personally read that for the group instead of election for the interface.

mobash-rasool commented:
Yeah @donaldsharp I too did the same mistake earlier. I fixed it for the group.

But then We found this issue while testing IGMPv2 conformance testing test case 3.10. I have fixed it and verified it. I will add the conformance ID in the commit msg to be clear now.
